File: 2017/5/src/main/java/com/mikeseagull/adventofcode2017/Part2.java
package com.mikeseagull.adventofcode2017;

import java.util.List;

public class Part2 extends Part1 {
    @Override
    public int main() {
        List<Integer> instructions = readFile("part2_input");
        int tracker = 0;
        int step_counter = 0;
        while (tracker < instructions.size()) {
            step_counter++;
            int original_tracker = tracker;
            int instruction = instructions.get(tracker);
            tracker += instruction;
            if (instruction >= 3) {
                instructions.set(original_tracker, instruction - 1);
            } else {
                instructions.set(original_tracker, instruction + 1);
            }
        }
        return step_counter;
    }
}

File: 2017/2/src/main/kotlin/com/mikeseagull/adventofcode2017/Part2.kt
package com.mikeseagull.adventofcode2017

import java.util.*
import kotlin.collections.ArrayList

class Part2: Part1() {
    fun evenlyDivisible(checkNumber: Int, numbers: ArrayList<Int>): Int {
        for (number in numbers) {
            if (number > checkNumber && number % checkNumber == 0) {
                //return Pair(number, checkNumber)
                return number / checkNumber
            } else if (checkNumber % number == 0) {
                return checkNumber / number
            }
        }
        var number = numbers.get(0)
        numbers.removeAt(0)
        return evenlyDivisible(number, numbers)
    }
    override fun main(): Int {
        var lines = readFile("/part2_input")

        var checksum = 0
        for (line in lines) {
            var numbers = parseLine(line)
            var number = numbers.get(0)
            numbers.removeAt(0)
            checksum = checksum + evenlyDivisible(number, numbers)
        }
        return checksum
    }
}